arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -2833,10 +2807,11 @@ Result IResearchAnalyzerFeature::storeAnalyzer(AnalyzerPool& pool) {
                   "' configuration while storage engine in recovery"};
     }
 
-    auto split = splitAnalyzerName(pool.name());
-    auto* vocbase = dbFeature.useDatabase(static_cast<std::string>(
-        split.first));  // FIXME: after C++20 remove cast and use heterogeneous
-                        // lookup
+    auto const split = splitAnalyzerName(pool.name());
+    // FIXME: after C++20 remove cast and use heterogeneous
+    // lookup
+    auto* vocbase =
```

Comment:
Is this `useDatabase()` call accompanied by some call to `vocbase->release()`? Otherwise I think this will only increase the reference count for the database, but never decrease it later.

Reply:
Thank you, I'll check

Reply:
Will be addressed in https://github.com/arangodb/arangodb/pull/17773